src/Uno.Extensions.Navigation.UI/Navigators/SelectorNavigator.cs

```diff
@@ -123,7 +123,9 @@ protected async Task SelectionChanged(FrameworkElement sender, FrameworkElement?
 			return;
 		}
 
-		await nav.NavigateRouteAsync(sender, path);
+		var data = selectedItem.GetData();
```

Comment:
@eriklimakc anything specifically that we could add to the docs for this? Or is there anything we would need to update if we have text somewhere saying that this wasn't previously possible?

Reply:
@kazo0 Yes, we can add an example [here](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/HowTo-NavigateInXAML.html#2-navigationdata) and remove the note that states this is not supported. Also we can change the [Navigation sample](https://github.com/unoplatform/Uno.Samples/blob/20a110a75de43c8ddb04f149ae96fe2b74c93af1/UI/Navigation/src/Navigation/Presentation/TabBarWithDataNavigation/TabBarWithDataViewModel.cs#L23-L29) in Uno.Samples to use the new approach.

Reply:
Logged an issue for the navigation sample https://github.com/unoplatform/Uno.Samples/issues/868